FeedbackForm component with emoji rating and comment input

Ruthesh123 opened the pull request:
- Implemented UI using React.createElement
- Includes 5 emoji-based rating buttons
- Adds optional comment textarea
- Stores feedback in localStorage
- Displays thank-you message on successful submission
- Tailwind CSS used for styling

Ruthesh123 committed:
FeedbackForm component with emoji rating and comment input

- Implemented UI using React.createElement
- Includes 5 emoji-based rating buttons
- Adds optional comment textarea
- Stores feedback in localStorage
- Displays thank-you message on successful submission
- Tailwind CSS used for styling

Ruthesh123 committed:
rewrite feedback from

- Replaced React.createElement with idiomatic JSX for clarity and maintainability
- Aligned component structure with project conventions and other existing components

Ruthesh123 commented:
Hi professor, Thanks for the feedback. I've refactored the FeedbackForm component to follow the existing codebase patterns using TypeScript with JSX (.tsx).

Ruthesh123 commented:
Hi Professor,
Kindly, Please review this PR

achow-conestogac commented:
hi Ruthesh, this is code code as the component is unused.

please implement this into the flow

Ruthesh123 committed:
Merge branch 'main' into FeedbackFormComponent

Ruthesh123 committed:
integrate FeedbackForm component into App layout

- Imported and rendered FeedbackForm below the main header in App.tsx
- Added a visually distinct section for user feedback using Tailwind styling
- Preserves original header design and structure
- Enhances user interaction by collecting emoji-based feedback

Ruthesh123 commented:
<img width="2188" height="1541" alt="Screenshot 2025-07-17 214501" src="https://github.com/user-attachments/assets/1787a07a-d648-4c75-bee5-52512c81a4d5" />

Ruthesh123 commented:
This FeedbackForm Component belongs to the Feedback Widget Module it is stand alone and it will sit at bottom of the mainscreen to take feedback from the user for the Overall Satisfaction with the Application, this is not related to the SuggestiveEvaluationModule.

I'm also working on WorkLoadAccuracyMeter which is a part of SuggestiveEvaluationModule. It is PR #523

Ruthesh123 committed:
Update Feedbackform for Better UI

Ruthesh123 committed:
Merge branch 'main' into FeedbackFormComponent

Ruthesh123 commented:
Thank you for your feedback, Professor. the previous version of the Feedback Widget was taking up too much vertical space and negatively impacting the user experience.

I've since refactored the component to sit unobtrusively at the bottom-right corner of the screen. The button now triggers a compact popup that appears directly above it, minimizing the space it occupies and ensuring it doesn't interfere with the main content of the application.

Ruthesh123 committed:
reposition to bottom-right corner with popup above button to improve UX

achow-conestogac commented on src/FeedbackForm/FeedbackForm.test.tsx:
```diff
@@ -0,0 +1,56 @@
+
+describe('FeedbackForm', () => {
+  beforeEach(() => {
+    localStorage.clear();
```
we have a local storage service. use that.

Ruthesh123 replied:
The new implementation uses local storage service

achow-conestogac commented on src/FeedbackForm/FeedbackForm.tsx:
```diff
@@ -0,0 +1,161 @@
+
+  return (
+    <div
+      style={{
+        position: 'fixed',
+        bottom: '20px',
+        right: '20px',
+        zIndex: 9999,
+        display: 'flex',
+        flexDirection: 'column',
+        alignItems: 'flex-end',
+      }}
```
you need to use tailwindcss for the styling.

Ruthesh123 replied:
Used tailwindcss for the styling.

achow-conestogac commented on src/FeedbackForm/FeedbackForm.tsx:
```diff
@@ -0,0 +1,161 @@
+          />
+
+          <div style={{ textAlign: 'right' }}>
+            <button
```
button should use the existing one from `src/component`

Ruthesh123 replied:
Used button existing one from src/component

achow-conestogac commented:
so who is this feedback information meant for and how do you get it there?

since the application is offline, how do these people get the data?

Ruthesh123 commented on src/App.tsx:
There is only Feedback Form, Professor.

Ruthesh123 committed:
Align feedback button to bottom-center and apply Tailwind styling

Ruthesh123 commented:
Hi professor, Thankyou for your feedback.
Updated FeedbackForm component layout to align the feedback button at the bottom center using TailwindCSS
Replaced inline styles with Tailwind utility classes as per feedback
Ensured popup closes when clicking outside and submits store data correctly
Fixed test file to use localStorage service and clear properly before each test

Ruthesh123 committed:
globalize FeedbackForm placement without changing App logic
